File: course-template/cpp/add_cpp/test_public.cpp
// Visible tests. Compiled together with add.cpp; a non-zero exit code fails the
// grading stage. Plain checks (not <cassert>) so the result is independent of
// NDEBUG / optimization flags.
#include <iostream>

int add(int a, int b);

static int check(int got, int want, const char* name) {
    if (got != want) {
        std::cerr << "FAIL " << name << ": got " << got << " want " << want << "\n";
        return 1;
    }
    return 0;
}

int main() {
    int fails = 0;
    fails += check(add(2, 3), 5, "simple_positive");
    fails += check(add(0, 0), 0, "zero");
    fails += check(add(-1, 1), 0, "mixed_signs");
    return fails == 0 ? 0 : 1;
}

File: course-template/go/add_go/test_public.go
// Visible tests. Compiled with add.go via `go run`; exits non-zero on mismatch.
// Named test_public.go (not *_test.go) so it is an ordinary main, not a `go test`
// file — the grader drives it explicitly.
package main

import (
	"fmt"
	"os"
)

func check(got, want int, name string) int {
	if got != want {
		fmt.Fprintf(os.Stderr, "FAIL %s: got %d want %d\n", name, got, want)
		return 1
	}
	return 0
}

func main() {
	fails := 0
	fails += check(Add(2, 3), 5, "simple_positive")
	fails += check(Add(0, 0), 0, "zero")
	fails += check(Add(-1, 1), 0, "mixed_signs")
	if fails != 0 {
		os.Exit(1)
	}
}
